pkg/test/dnszone/dnszone.go

```diff
@@ -52,6 +51,17 @@ func WithControllerOwnerReference(owner metav1.Object) Option {
 	return Generic(generic.WithControllerOwnerReference(owner))
 }
 
+// WithControllerOwnerReferenceAPIVersion sets the api version on the controller owner reference.
+func WithControllerOwnerReferenceAPIVersion(apiVersion string) Option {
```

Comment:
This function is very specific to the ownership tests. Can we leave it in that file rather than adding it here?

Reply:
Sure, but I reserve the right to put it back if I end up needing it in a different package. :)